src/System.ComponentModel.TypeConverter/src/System/ComponentModel/MemberDescriptor.cs

```diff
@@ -421,18 +422,24 @@ private void FilterAttributesIfNeeded()
                 }
                 else
                 {
-                    list = new ArrayList(_attributes);
+                    list = new List<Attribute>(_attributes);
                 }
 
-                Hashtable hash = new Hashtable(list.Count);
+                var set = new HashSet<object>();
 
-                foreach (Attribute attr in list)
+                for (int i = 0; i < list.Count;)
                 {
-                    hash[attr.GetTypeId()] = attr;
+                    if (set.Add(list[i].GetTypeId()))
+                    {
+                        ++i;
+                    }
+                    else
+                    {
+                        list.RemoveAt(i);
+                    }
                 }
```

Comment:
This code already doesn't appear to be particularly optimized. If the library is already referencing System.Linq, seems reasonable to use it, assuming performance here isn't critical. For now, though, I'd suggest leaving it as is, to get the behavioral fix in place.